server/conf/evolutions/default/67.sql

```diff
@@ -0,0 +1,11 @@
+# --- !Ups
+CREATE TABLE IF NOT EXISTS application_statuses (
+                                                  id BIGSERIAL PRIMARY KEY NOT NULL,
+                                                  program_name VARCHAR NOT NULL,
```

Comment:
"No, a foreign key does not have to be the primary key in another table, but it does need to refer to a primary key or a unique key in the parent table. " Our program_name is neither. Hence this will fail.

Reply:
We could have the repository fail the transaction if the program doesn't exist